test/unit/core/detail/pack_algorithm_test.cpp

```diff
@@ -7,13 +7,58 @@
 
 #include <gtest/gtest.h>
 
+#include <sstream>
+
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
-#include <seqan3/core/algorithm/parameter_pack.hpp>
+#include <seqan3/core/detail/pack_algorithm.hpp>
 #include <seqan3/core/type_list/type_list.hpp>
 
 using namespace seqan3;
 
-TEST(parameter_pack, for_each_value)
+//-----------------------------------------------------------------------------
+// all_of
+//-----------------------------------------------------------------------------
+
+struct test_integral_meta
+{
+    bool operator()(...) { return false; }
+
+    template <typename identity_t>
+        requires std::integral<typename identity_t::type>
+    bool operator()(identity_t) { return true; }
+};
+
+auto test_integral_value = [] (auto value)
+{
+    return std::is_integral_v<decltype(value)>;
+};
+
+TEST(pack_algorithm, all_of_in_type_list)
+{
+    EXPECT_TRUE(detail::all_of<type_list<>>(test_integral_meta{}));
+    EXPECT_TRUE((detail::all_of<type_list<int8_t, int16_t, uint32_t>>(test_integral_meta{})));
```

Comment:
```suggestion
    EXPECT_TRUE((detail::all_of<type_list<int8_t, int16_t, uint32_t>>(is_integral_fn{})));
```

I think it is easier to read if you write test as if you would actually use the API. The test here is not for `test_integral_meta`, but for `all_of`. So the name should not contain test.